### PR TITLE
[PURCHASE-2574]: Swipe down dismiss modal

### PR DESCRIPTION
**Pull request by xtina-starr**
paired with: @lilyfromseattle @ds300
The type of this PR is: **FEAT**

This PR resolves [PURCHASE-2574]

### Description
This is follow up to https://github.com/artsy/eigen/pull/4648 and adds a new native modal dismissed event and uses that in the conversation to determine when we need to refetch conversation data. Previously we were hooking into the local modalDismissed and goBack events in eigen. With this implementation those are no longer needed.

https://user-images.githubusercontent.com/5201004/117042864-7f171800-acda-11eb-8cc6-a5170de842ec.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[PURCHASE-2574]: https://artsyproduct.atlassian.net/browse/PURCHASE-2574

### CODE REVIEW

File: src/lib/navigation/navigate.ts [hunk 1]
```diff
@@ -76,12 +76,10 @@ export const navigationEvents = new EventEmitter()
 
 export function dismissModal() {
   LegacyNativeModules.ARScreenPresenterModule.dismissModal()
-  navigationEvents.emit("modalDismissed")
 }
```
Comment (xtina-starr) [inline src/lib/navigation/navigate.ts:79]: We were able to remove this in favor of emitting the `modalDismissed` event inside of the NativeModel file.

File: src/lib/navigation/navigate.ts [hunk 2]
```diff
@@ -76,12 +76,10 @@ export const navigationEvents = new EventEmitter()
...
 
 export function goBack() {
   LegacyNativeModules.ARScreenPresenterModule.goBack(unsafe__getSelectedTab())
-  navigationEvents.emit("goBack")
```
Comment (xtina-starr) [inline src/lib/navigation/navigate.ts:84]: This also get taken care of by the native `modalDismissed` event.

File: src/lib/Scenes/Inbox/Screens/Conversation.tsx [hunk 3]
```diff
@@ -97,7 +95,16 @@ export class Conversation extends React.Component<Props, State> {
   }
 
   refetch = () => {
-    this.props.relay.refetch({})
+    this.props.relay.refetch(
+      { conversationID: this.props.me.conversation?.internalID },
+      null,
+      (error) => {
+        if (error) {
+          console.error("Conversation.tsx", error.message)
```
Comment (erikdstock) [inline src/lib/Scenes/Inbox/Screens/Conversation.tsx:103]: Is this new behavior?

**Review by erikdstock — COMMENTED** [review]
Looks good from here. Do we need any equivalent android changes?

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Nice one!

**Comment by MounirDhahri** [comment]
About @erikdstock comment on android, you may need to use react-navigation event listeners (https://reactnavigation.org/docs/navigation-events/) or `useIsFocused` (https://reactnavigation.org/docs/use-is-focused/) to know if the user dismissed the modal

**Review by ds300 — APPROVED** [review]

**Comment by ds300** [comment]
re: android. It's not possible to swipe-dismiss modals on android so we can also just keep the event emit statements in `navigate.ts`. i.e. the ones in `goBack` and `dismissModal` that this PR currently is removing.

```diff
 export function dismissModal() {
   LegacyNativeModules.ARScreenPresenterModule.dismissModal()
-  navigationEvents.emit("modalDismissed")
+  if (Platform.OS === "android") {
+    navigationEvents.emit("modalDismissed")
+  }  
 }
 
 export function goBack() {
   LegacyNativeModules.ARScreenPresenterModule.goBack(unsafe__getSelectedTab())
-  navigationEvents.emit("goBack")
+  if (Platform.OS === "android") {
+    navigationEvents.emit("goBack")
+  }  
 }
```

**Review by erikdstock — APPROVED** [review]

**Comment by MounirDhahri** [comment]
@xtina-starr I will merge this for now but do you mind if you create a new ticket on your board with the follow up items

**Comment by MounirDhahri** [comment]
This is good to go for now I guess! So let's merge it 
@xtina-starr do you mind creating a ticket on your board with the follow up items that David suggested and linking his comment maybe as a reference. Thank you

**Comment by xtina-starr** [comment]
@MounirDhahri yes, will do. Thanks for merging 🙏🏿

**Pull request merged**